SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+#ifndef SMALL_BETTER_LOGGER_H
+#define SMALL_BETTER_LOGGER_H
+
+// Used for writing to output stream
+#include <iostream>
+#include <fstream>
+
+// Used for formatting and creating the output string
+#include <sstream>
+#include <vector>
+#include <regex>
+
+namespace sbl
+{
+	// Logger declaration
+	class Logger;
+	using logger = Logger;
+
+	// Stream types to be used by a Logger instance
+	enum class STREAM_TYPE
+	{
+		STDOUT, STDERR, STDLOG, FILE
+	};
+	using stream_type = STREAM_TYPE;
+
+	// Logger class: The main part of the library, used to log messages to a stream (ex.: STDOUT, STDERR, file etc.)
+	class Logger
+	{
+	public:
+		// Constructors and destructors
+
+		// Creates an instance of Logger which outputs to a stream chosen from a STREAM_TYPE (except STREAM_TYPE::FILE)
+		// Defaults to STREAM_TYPE::STDOUT and no auto flush
+		Logger(STREAM_TYPE type = STREAM_TYPE::STDOUT)
```

Comment:
I feel like letting the user choose the FILE stream type and throwing an exception is not really the way to go with a library. Given that you have a separte fileStream member and the only constructor that enables file logging takes a path and all others throw, I would suggest an abstract base class Logger, and 2 derived classes: the FileLogger which hold the file stream and the StreamLogger which holds STREAM_TYPE enum, just that now FILE is not a member of the enum.